rhaptos/cnxmlutils/xsl/cnxml-to-html5.xsl

```diff
@@ -275,7 +275,7 @@
 
 <!-- Help ensure that HTML paragraphs do not contain blockish elements as children -->
 
-<xsl:template match="c:para[not(.//c:figure|.//c:list[not(@display='inline')]|.//c:table|.//c:media[not(@display) or @display='block']|.//c:equation|.//c:preformat|.//c:note|.//c:exercise)]" name="convert-para">
+<xsl:template match="c:para[not(.//c:figure|.//c:list[not(@display='inline')]|.//c:table|.//c:media[not(@display) or @display='block']|.//c:equation[not(ancestor::c:footnote)]|.//c:preformat|.//c:note|.//c:exercise)]" name="convert-para">
```

Comment:
I'm not sure this is resilient to a structure like: `footnote -> para -> equation` even though it's not intended for that case. Maybe theres a crafty xpath selector that ensures the footnote is a child of the para as well to enforce a `para -> footnote -> equation`-like structure?

Reply:
@tomjw64 - Is this scenario you are referring to?:

```
  <footnote>
    <para>
      <equation>x=3</equation>
    </para>
  </footnote>
```

In the above, that equation will get unwrapped [here](https://github.com/openstax/rhaptos.cnxmlutils/blob/9d78f4f1dfbde353e9c37a556573b862df2fff26/rhaptos/cnxmlutils/xsl/cnxml-to-html5.xsl#L952-L958) and you end up with the `<aside>` being pulled down to the end since it's not part of a `<para>`.

Reply:
Ah I see. So it's very unlikely that a footnote could be an ancestor outside of the para of interest. That's probably good enough.

Reply:
Oh, in case you mean this scenario:

```
  <para>
    Content with footnote
    <footnote id="p5">
     <para>
      Footnote with equation: <equation id="eq1" class="unnumbered">x=2</equation>
      </para>
    </footnote>
  </para>
```

That works as well due to the unwrapping (produces the same output within the test case I added except for the comment on unwrapping). There probably is still some corner case, though, with the just the right mix of things.